blog/modules/ROOT/pages/sort-antora-components.adoc

```diff
@@ -0,0 +1,196 @@
+= Awesome Antora: Sort and hide components and versions
+David Jencks <djencks@apache.org>
+:page-revdate: 2020-9-28 14:25
+:page-author: David Jencks
+:description: You would like custom sorting of Antora components and versions, or to hide components.
+:page-tags: Antora, Sorting Components
+:page-image: sort.jpg
+:page-color-primary: #8BB3B2
+:page-color-secondary: #C68B45
+:ms-sorter-url: https://gitlab.com/opendevise/oss/antora-site-generator-ms/-/blob/v2.3.x/lib/custom-sort-components.js
+
+A frequent request is to sort the components and their versions in the default Antora UI component explorer in a customized order.
+Fortunately, this is quite easy to do using a couple of files in a supplemental UI or in a customized UI.
+I'd like to thank Bob Roro and Ewan Edwards for pushing me into figuring out how this works and a lot of hints and answers and Dan Allen for pointing to an link:{ms-sorter-url}[existing implementation] (for a completely custom UI).
+
+== Where do we sort?
+
+Here's the default UI's partial that sets up the component explorer contents:
+
+.nav-explore.hbs (original)
+[source,handlebars]
+----
+include::example$sort-components/nav-explore-default.hbs[]
+----
+<1> The crucial line!
+
+Here, `{{#each site.components}}` iterates over the object `site.components` in the default order.
+All we have to do is use a javascript helper to change the order, as in `{{#each (mySorter site.components)}}`.
+For this the javascript helper signature would be `module.exports = (components) => {...}`.
+Note that `#each` works on either arrays or objects, so, although `site.components` is an object, we can return an array of components in the desired order.
+
+Now lets consider plausible sources of information for the sorter.
+First of all, we need the collection of components.
+This is actually available in two ways, as `site.components` or by calling `getComponents` on the content catalog model as `site.contentCatalog`.
+We could just pass the entire `site` model to our helper, but lets just use `site.components` like the default.
+
+This much is adequate to hard-code a sort, for instance by listing the order of components explicitly.
+
+Lets start with a possibly silly example that demonstrates a sorting algorithm that does not need to be configured.
+
+Here's the `nav-explore.hbs` modified to work with a sorter:
+
+.nav-explore.hbs (with automonous component sorting)
+[source,handlebars]
+----
+include::example$sort-components/nav-explore-components-autonomous.hbs[]
+----
+
+
+.componentSorter.js (Reverse sorting)
+[source,javascript]
+----
+include::example$sort-components/componentSorter-autonomous.js[]
+----
+
+Here we just reverse the original sort order.
```

Comment:
```suggestion
Here we reverse the original sort order.
```

Reply:
How about `Here, we simply reverse the original sort order.`?

Reply:
I believe that it's a good practice to avoid using "simply" and "just". But maybe in this context, it's fine because it's not really a procedure that the user must execute.

Reply:
Perhaps you're right... how about

Here, all we've done is to reverse the original sort order.

Reply:
I like it! :100: